lib/chef/chef_fs/file_system/repository/chef_repository_file_system_root_dir.rb

```diff
@@ -109,7 +110,14 @@ def create_child(name, file_contents = nil)
             else
               child_paths[name].each do |path|
                 begin
-                  Dir.mkdir(path)
+                  events = Chef::EventDispatch::Dispatcher.new
+                  node = Chef::Node.new
+                  run_context = Chef::RunContext.new(node, {}, events)
+                  dir = Chef::Resource::Directory.new(path, run_context)
```

Comment:
and all of this framework to build a resource is an indication that these objects aren't well constructed to be reusable like this, so shouldn't be used here.